src/Sarif.Viewer.VisualStudio/Sarif/Result.extensions.cs

```diff
@@ -60,10 +60,12 @@ public static string GetPrimaryTargetFile(this Result result)
             {
                 return primaryLocation.AnalysisTarget.Uri.ToPath();
             }
-            else
+            else if (primaryLocation.FullyQualifiedLogicalName != null)
```

Comment:
Some files have no ResultFile, AnalysisTarget or FullyQualifiedLogicalName.  In this case, return an empty string for the file name.  Must return some value other than null because this is used as a key in a dictionary.